webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -1951,3 +1953,169 @@ func (v validator) validateGroupName(
+					allErrs = append(allErrs, field.NotSupported(
+						p.Child("topologyKey"), rs.TopologyKey, []string{topology.KubernetesTopologyZoneLabelKey}))
+				}
+			}
+		}
+
+		if len(a.PreferredDuringSchedulingIgnoredDuringExecution) > 0 {
+			p := p.Child("preferredDuringSchedulingIgnoredDuringExecution")
+
+			for idx, rs := range a.PreferredDuringSchedulingIgnoredDuringExecution {
+				p := p.Index(idx)
+
+				if rs.LabelSelector != nil {
+					p := p.Child("labelSelector")
+
+					for exprIdx, expr := range rs.LabelSelector.MatchExpressions {
+						p := p.Child("matchExpressions").Index(exprIdx)
+
+						if expr.Operator != metav1.LabelSelectorOpIn {
+							allErrs = append(allErrs, field.NotSupported(
+								p.Child("operator"),
+								expr.Operator,
+								[]metav1.LabelSelectorOperator{metav1.LabelSelectorOpIn}))
+						}
+					}
+				}
+
+				if rs.TopologyKey != topology.KubernetesTopologyZoneLabelKey {
+					allErrs = append(allErrs, field.NotSupported(
+						p.Child("topologyKey"), rs.TopologyKey, []string{topology.KubernetesTopologyZoneLabelKey}))
+				}
+			}
+		}
+	}
+
+	if a := affinity.VMAntiAffinity; a != nil {
+		p := path.Child("vmAntiAffinity")
+
+		if len(a.RequiredDuringSchedulingIgnoredDuringExecution) > 0 {
+			allErrs = append(allErrs, field.Forbidden(p.Child("requiredDuringSchedulingIgnoredDuringExecution"),
+				"VM anti-affinity with RequiredDuringSchedulingIgnoredDuringExecution is not allowed"))
+		}
+
+		if len(a.PreferredDuringSchedulingIgnoredDuringExecution) > 0 {
+			p := p.Child("preferredDuringSchedulingIgnoredDuringExecution")
+
+			for idx, rs := range a.PreferredDuringSchedulingIgnoredDuringExecution {
+				p := p.Index(idx)
+
+				if rs.LabelSelector != nil {
+					p := p.Child("labelSelector")
+
+					for exprIdx, expr := range rs.LabelSelector.MatchExpressions {
+						p := p.Child("matchExpressions").Index(exprIdx)
+
+						if expr.Operator != metav1.LabelSelectorOpIn {
+							allErrs = append(allErrs, field.NotSupported(
+								p.Child("operator"),
+								expr.Operator,
+								[]metav1.LabelSelectorOperator{metav1.LabelSelectorOpIn}))
+						}
+					}
+				}
+
+				if rs.TopologyKey != topology.KubernetesTopologyZoneLabelKey {
+					allErrs = append(allErrs, field.NotSupported(
+						p.Child("topologyKey"), rs.TopologyKey, []string{topology.KubernetesTopologyZoneLabelKey}))
+				}
+			}
+		}
+
+		if len(a.RequiredDuringSchedulingPreferredDuringExecution) > 0 {
+			allErrs = append(allErrs, field.Forbidden(p.Child("requiredDuringSchedulingPreferredDuringExecution"),
+				"VM anti-affinity with RequiredDuringSchedulingPreferredDuringExecution is not allowed"))
+		}
+
+		if len(a.PreferredDuringSchedulingPreferredDuringExecution) > 0 {
+			p := p.Child("preferredDuringSchedulingPreferredDuringExecution")
+
+			for idx, rs := range a.PreferredDuringSchedulingPreferredDuringExecution {
+				p := p.Index(idx)
+
+				if rs.LabelSelector != nil {
```

Comment:
This also needs to be updated to allow MatchExpressions, but only `In` operator.